include/igl/blkdiag.cpp

```diff
@@ -0,0 +1,77 @@
+// This file is part of libigl, a simple c++ geometry processing library.
+// 
+// Copyright (C) 2020 Alec Jacobson <alecjacobson@gmail.com>
+// 
+// This Source Code Form is subject to the terms of the Mozilla Public License 
+// v. 2.0. If a copy of the MPL was not distributed with this file, You can 
+// obtain one at http://mozilla.org/MPL/2.0/.
+#include "blkdiag.h"
+
+template <typename Scalar>
+IGL_INLINE void igl::blkdiag(
+  const std::vector<Eigen::SparseMatrix<Scalar>> & L, 
+  Eigen::SparseMatrix<Scalar> & Y)
+{
+  int nr = 0;
+  int nc = 0;
+  int nnz = 0;
+  for(const auto & A : L)
+  {
+    nr += A.rows();
+    nc += A.cols();
+  }
+  Y.resize(nr,nc);
+  Eigen::VectorXi per_col = Eigen::VectorXi::Zero(nc);
+  for(int j = 0;j<L.size();j++)
+  {
+    per_col(j) = L[j].cols();
+  }
+  Y.reserve(per_col);
```

Comment:
Hmm, this is wrong. You're reserving a number of nonzero equal to the #cols in each block matrix, which is overkill, but also the indices don't match (`L.size()` is < `nc`).

You should count the number of nnz using `per_col(c + offset) = L[j].innerVector(c).nonZeros();`.

Reply:
right. this was wrong. deleted for now.